### ISSUE TITLE
pseudo-NVRAM doesn't flush on EFI application/loader image loading another EFI app/loader image

### ISSUE THREAD

**Issue by esnowberg** [andreiw/RaspberryPiPkg#47]
If I use one of the pre-built images the pseudo-NVRAM works fine.  However, if I follow the build instructions within the readme.   The new RPI_EFI.fd image I built works, with the exception of the pseudo-NVRAM.  Are there additional build steps to get that to work?

**Comment by andreiw** [comment]
What are the exact steps and what behavior are you seeing exactly? An attached serial log from a debug build will be useful.

A

**Comment by esnowberg** [comment]
I'm trying to set the System BootOrder and add a Boot Entry.  It will persist with your prebuilt image.  But it doesn't with the one I built.

Debug log attached.
[nvram-problem.txt](https://github.com/andreiw/RaspberryPiPkg/files/2168835/nvram-problem.txt)

**Comment by andreiw** [comment]
Fun.

- Are there any other media present that may have an RPI_EFI.FD file on them?
- Can you build using GCC 4.9 or GCC5 (I know, I know..)
- After creating the boot entry, what happens when you select "reset the system" (can you give me a full log with that?)

**Comment by esnowberg** [comment]
There isn't any other media present.  The only thing connected is the micro sd card and the serial cable.

The BOOTAA64.EFI program is really fbaa64.efi, so it creates the boot entries and then jumps to shim. This time once shim loaded GRUB2  Instead of booting into Linux, I rebooted from the GRUB2 shell.  This brought me back to UEFI.  I was then going to reset the system.  But first I looked at the Boot Manager Menu and saw my variable set.  Instead of resetting the system, I just booted into Linux.  Now the two variables persist.

To reproduce the problem if I do:
UEFI -> BOOTAA64.EFI (which sets variables) -> shim -> GRUB2 -> Linux
The variables created do not persist.

If I do:
UEFI -> BOOTAA64.EFI (which sets variables) -> shim -> GRUB2 -> reboot 
The variables persist.
